pkg/sql/serial.go

```diff
@@ -211,6 +213,23 @@ func (p *planner) generateSerialInColumnDef(
 			tree.SequenceOption{Name: tree.SeqOptCache, IntVal: &value},
 		}
 	}
+
+	// Setup the type of the sequence based on the type observed within
+	// the column.
+	switch asIntType {
+	case types.Int2, types.Int4:
+		// Valid types, nothing to do.
+	case types.Int:
+		// Int is the default, so no cast necessary.
```

Comment:
does this correctly handle the case where `default_int_size` is changed?

Reply:
Ah, have to handle that case. Let me adjust the code